Makefile

```diff
@@ -58,8 +58,13 @@ checksum:
 clean:
 	@rm -rf $(BINDIR) ./_dist
 
+GIT_BASEDIR    = $(shell git rev-parse --show-toplevel 2>/dev/null)
+ifneq ($(GIT_BASEDIR),)
+	LDFLAGS += -X github.com/Azure/acs-engine/pkg/test.JUnitOutDir=${GIT_BASEDIR}/test/junit
+endif
+
 test: test-style
-	go test -v $(GOFILES)
+	go test -v -ldflags '$(LDFLAGS)' $(GOFILES)
```

Comment:
I believe you can use `gingko` to run unit or e2e BDD-style tests. For running only unit tests, pass `$GOFILES` to ginkgo.